gen-ai/orchestrator-core/src/main/kotlin/ai/tock/genai/orchestratorcore/utils/SecurityUtils.kt

```diff
@@ -83,5 +83,31 @@ object SecurityUtils {
             secretMangerService.createOrUpdateSecretKey(namespace, botId, feature, secretValue)
         } ?: RawSecretKey(secretValue)
 
+    /**
+     * Delete a secret
+     * @param secret the secret key
+     */
+    fun deleteSecret(secret: SecretKey) {
+        try {
+            // If the secret is a raw value, there's nothing to be done
+            if(secret is RawSecretKey) return
+
+            // Check SecretManagerProvider if it is defined
+            if (genAISecretManagerProvider == null) {
+                throw IllegalArgumentException("No Gen AI secret manager provider has been defined to delete the secret. Type=${secret.type}")
+            }
+
+            // Check whether the SecretManagerProvider supports secret
+            if(secretMangerService.isSecretTypeSupported(secret)) {
+                return secretMangerService.deleteSecret((secret as NamedSecretKey).secretName)
+            }else{
+                throw IllegalArgumentException("The secret manager provider type '${secret.type}' is not supported by " +
+                        "the instantiated service ${secretMangerService::class.simpleName}.")
+            }
+        } catch (e: Exception) {
+            logger.warn("The secret has not been deleted.", e)
+            // Do not block treatment if it fails.
+        }
```

Comment:
Very good point for the non-blocking failure. So why theses immediately caught exceptions ?
I think the exception handling should be made in higher layers, letting the consumer decide what to do in case of failure, depending on the context. WDYT ?

Reply:
Yes, you're absolutely right, but today on tock we're using this service in 10 places, and they're all iso in terms of behavior. In fact, that's exactly what we did for `fetchSecretKeyValue`.